### PR TITLE
Rating movies

### PR DESCRIPTION
**Pull request by ckoga**
### What's this PR do?
-This PR builds out user rating functionality
-creates an apiCalls file
-Changes the user info sent to the Redux store

### Where should the reviewer start?
-Look the Login component and see that it has been converted to a class component
-look over fetch calls that start in apiCalls.js and update the redux store in Login

### How should this be manually tested?
-Pull down this branch and check the redux store
-once you log in check the user ratings state
-pick a movie to rate then check the store again to see updated userRatings

### CODE REVIEW

File: rancid-tomatillos/src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const fetchUser = (options) => {
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/apiCalls.js:1]: Nice work starting to get the fetches pulled out into a seperate file to make them reusable and easier to test!

File: rancid-tomatillos/src/containers/Login/Login.js [hunk 2]
```diff
@@ -29,17 +30,16 @@ class Login extends Component {
...
-        console.log(data)
-        this.props.addUser(data)
+        this.props.addUser(data.user.id)
         this.props.updateLoggedIn(this.props.isLoggedIn)
+        return data.user.id
+      })
+      .then(id => {
+        fetchRatings(id)
+          .then(data => this.props.updateUserRatings(data.ratings))
+          .catch(err => console.log(err))
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/Login/Login.js:42]: Should we have an error property in our redux store to update on all of our catches??

File: rancid-tomatillos/src/containers/MoviePage/MoviePage.js [hunk 3]
```diff
@@ -1,33 +1,99 @@
...
-            <article className='movie-page-details'>
-              <h2>Rating</h2>
-              <h2>{average_rating}</h2>
-            </article>
-          </section>
-      </section>
-    </main>
-  )
+
+class MoviePage extends Component {
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/MoviePage/MoviePage.js:9]: good work switching this to a class based component to have a controlled form for the user rating input. This looks like it also made it cleaner being able to have our fetch/post as a method on this component.

File: rancid-tomatillos/src/containers/MoviePage/MoviePage.js [hunk 4]
```diff
@@ -1,33 +1,99 @@
...
+  }
+
+  render() {
+    const { id, movies, isLoggedIn } = this.props
+
+    let movie = movies.movies.find(movie => movie.id === parseInt(id));
+    const { average_rating, backdrop_path, overview, poster_path, release_date, title } = movie;
+    const rating =
+    <> 
+      <h2>Rate This Movie</h2>
```
Comment (adamsjr8576) [inline rancid-tomatillos/src/containers/MoviePage/MoviePage.js:50]: I think if we have time we could try and style this and make it look fancier. Maybe something like one of those wheel scroll windows or something like that: I think we could take a look at this plugin - looks pretty simple to implement and has a good 1/10 option! https://antennaio.github.io/jquery-bar-rating/

**Review by adamsjr8576 — APPROVED** [review]
Nice work implementing the functionality of having a user once logged in being able to rate a movie! This looks good to merge to me

File: rancid-tomatillos/src/containers/MoviePage/MoviePage.js (hunk 4 above)
Comment (JamesRexMiller4) [reply rancid-tomatillos/src/containers/MoviePage/MoviePage.js:50]: I am game! Can you send a snippet of what it looks like or a gif?

**Pull request merged**